src/core/image_processor.py

```
@@ -211,353 +212,415 @@ def detect_neurons(
-            image_gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
+        import tifffile
+        from PIL import Image
+        
+        path = Path(img_path)
+        suffix = path.suffix.lower()
+        
+        # Use tifffile for TIFF files (supports 16-bit, multi-page, etc.)
+        if suffix in ['.tif', '.tiff']:
+            try:
+                img = tifffile.imread(str(path))
+            except Exception as e:
```

Ruff (0.14.3)

Do not catch blind exception: `Exception`

(BLE001)